TestFleet* fail on 8.6.x

When updating the stack version to 8.6.0 (#6327), `TestFleetKubernetesIntegrationRecipe` fails.

https://devops-ci.elastic.co/job/cloud-on-k8s-pr-e2e-tests/98/

```
=== RUN   TestFleetKubernetesIntegrationRecipe/ES_data_should_pass_validations
Retries (30m0s timeout): .......
    step.go:44: 
        Error: Received unexpected error:
            elasticsearch client failed for https://elasticsearch-6qxw-es-internal-http.e2e-nru4o-mercury.svc:9200/_data_stream/logs-elastic_agent.fleet_server-default: 404 Not Found: 
            RootCause:[{
                Reason: no such index [logs-elastic_agent.fleet_server-default]
                Type: index_not_found_exception
             }]
    --- FAIL: TestFleetKubernetesIntegrationRecipe/ES_data_should_pass_validations (1800.00s)
```

Index `logs-elastic_agent.fleet_server-default`is missing because fleet server and agents are in error.  

I can reproduce the issue just by applying [`config/recipes/elastic-agent/fleet-kubernetes-integration.yaml`](https://github.com/elastic/cloud-on-k8s/blob/main/config/recipes/elastic-agent/fleet-kubernetes-integration.yaml) after updating the stack version to `8.6.0`.

The fleet-server agent pod is running, but the log shows that the process terminated as soon as it was enrolled.

```
> k logs fleet-server-agent-6b8bb8976f-hcz24 | grep '{' | tail -15 | jq .message -r
fleet-server routes set up
server hard connection limiter installed
Running on policy with Fleet Server integration: eck-fleet-server; missing config fleet.agent.id (expected during bootstrap process)
Unit state changed fleet-server-default-fleet-server (STARTING->DEGRADED): Running on policy with Fleet Server integration: eck-fleet-server; missing config fleet.agent.id (expected during bootstrap process)
Unit state changed fleet-server-default (STARTING->DEGRADED): Running on policy with Fleet Server integration: eck-fleet-server; missing config fleet.agent.id (expected during bootstrap process)
Fleet Server - Running on policy with Fleet Server integration: eck-fleet-server; missing config fleet.agent.id (expected during bootstrap process)
Starting enrollment to URL: https://fleet-server-agent-http.default.svc:8220/
Elastic Agent successfully enrolled
Elastic Agent successfully enrolled
Elastic Agent has been enrolled; start Elastic Agent
signal "terminated" received
Shutting down Elastic Agent and sending last events...
On signal
Stopping
error while closing server
```

Because fleet-server is in error, 2 agent pods are in `CrashLoopBackOff` state and the third is stuck retrying to reach fleet-server:

```
NAME                                  READY   STATUS             RESTARTS        AGE
elastic-agent-agent-8cc82             0/1     CrashLoopBackOff   9 (4m35s ago)   27m
elastic-agent-agent-8mm5w             1/1     Running            1 (27m ago)     27m
elastic-agent-agent-fgrnn             0/1     CrashLoopBackOff   9 (5m ago)      27m
```

```
> k logs elastic-agent-agent-8cc82 # (CrashLoopBackOff)
Updating certificates in /etc/ssl/certs...
rehash: warning: skipping ca-certificates.crt,it does not contain exactly one certificate or CRL
1 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
{"log.level":"info","@timestamp":"2023-01-17T14:15:47.237Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":475},"message":"Starting enrollment to URL: https://fleet-server-agent-http.default.svc:8220/","ecs.version":"1.6.0"}
Error: fail to enroll: fail to execute request to fleet-server: dial tcp 10.111.196.28:8220: connect: connection refused
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.6/fleet-troubleshooting.html
Error: enrollment failed: exit status 1
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.6/fleet-troubleshooting.html
```

```
> k logs elastic-agent-agent-8mm5w | tail -1 # (Running)
{"log.level":"error","@timestamp":"2023-01-17T14:11:39.924Z","log.origin":{"file.name":"fleet/fleet_gateway.go","file.line":194},
  "message":"Cannot checkin in with fleet-server, retrying",
  "error":{"message":"fail to checkin to fleet-server: all hosts failed: 1 error occurred:\n\t* requester 0/1 to host https://fleet-server-agent-http.default.svc:8220/ errored: Post \"https://fleet-server-agent-http.default.svc:8220/api/fleet/agents/996ab694-ef86-4f09-9ad3-47b7cf2fa1ed/checkin?\": dial tcp 10.111.196.28:8220: connect: connection refused\n\n"},"request_duration_ns":5733281,"failed_checkins":5,"retry_after_ns":739653326754,"ecs.version":"1.6.0"}
```

<details>
<summary>Full log:</summary>

```
Updating certificates in /etc/ssl/certs...
rehash: warning: skipping ca-certificates.crt,it does not contain exactly one certificate or CRL
1 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
Policy selected for enrollment:  eck-fleet-server
info 2023-01-17T13:47:58.184Z origin":{"file.name":"cmd/enroll_cmd.go file.line":594},"message":"Spawning Elastic Agent daemon as a subprocess to complete bootstrap process. ecs.version":"1.6.0"}
info 2023-01-17T13:47:58.963Z origin":{"file.name":"cmd/run.go file.line":188},"message":"APM instrumentation disabled ecs.version":"1.6.0"}
info 2023-01-17T13:47:59.040Z origin":{"file.name":"application/application.go file.line":44},"message":"Gathered system information ecs.version":"1.6.0"}
info 2023-01-17T13:47:59.151Z origin":{"file.name":"application/application.go file.line":50},"message":"Detected available inputs and outputs inputs":["syslog mongodb/metrics cloudfoundry/metrics nats/metrics prometheus/metrics osquery aws-cloudwatch mqtt uwsgi/metrics haproxy/metrics kafka/metrics sql/metrics cloudbeat/cis_k8s postgresql/metrics oracle/metrics elasticsearch/metrics unix windows/metrics syncgateway/metrics azure-eventhub mssql/metrics iis/metrics aws/metrics cel tcp nginx/metrics stan/metrics traefik/metrics aws-s3 synthetics/tcp system/metrics cloudbeat httpjson cloudbeat/cis_eks synthetics/icmp beat/metrics apache/metrics packet container netflow udp azure/metrics jolokia/metrics journald gcp-pubsub redis fleet-server containerd/metrics activemq/metrics audit/auditd awsfargate/metrics vsphere/metrics mysql/metrics audit/system kafka winlog synthetics/http docker/metrics http/metrics gcp/metrics audit/file_integrity log synthetics/browser logstash/metrics rabbitmq/metrics endpoint cloudfoundry http_endpoint o365audit filestream kibana/metrics zookeeper/metrics apm kubernetes/metrics linux/metrics redis/metrics etcd/metrics docker"],"ecs.version":"1.6.0"}
info 2023-01-17T13:47:59.152Z origin":{"file.name":"capabilities/capabilities.go file.line":54},"message":"Capabilities file not found in /usr/share/elastic-agent/capabilities.yml ecs.version":"1.6.0"}
info 2023-01-17T13:47:59.152Z origin":{"file.name":"application/application.go file.line":56},"message":"Determined allowed capabilities ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.056Z origin":{"file.name":"application/application.go file.line":105},"message":"Parsed configuration and determined agent is in Fleet Server bootstrap mode ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.058Z log.logger":"api origin":{"file.name":"api/server.go file.line":68},"message":"Starting stats endpoint ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.058Z log.logger":"api origin":{"file.name":"api/server.go file.line":70},"message":"Metrics endpoint listening on: 127.0.0.1:6791 (configured: http://localhost:6791) ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.143Z log.logger":"composable.providers.kubernetes origin":{"file.name":"kubernetes/kubernetes.go file.line":113},"message":"Kubernetes provider started for resource pod with node scope ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.144Z origin":{"file.name":"upgrade/upgrade.go file.line":102},"message":"Source URI reset from \"https://artifacts.elastic.co/downloads/\" to \"https://artifacts.elastic.co/downloads/\" ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.241Z log.logger":"composable.providers.docker origin":{"file.name":"docker/docker.go file.line":44},"message":"Docker provider skipped, unable to connect: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running? ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.244Z origin":{"file.name":"cmd/enroll_cmd.go file.line":770},"message":"Waiting for Elastic Agent to start Fleet Server ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.246Z origin":{"file.name":"coordinator/coordinator.go file.line":729},"message":"Updating running component model ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.343Z log.logger":"composable.providers.kubernetes origin":{"file.name":"kubernetes/util.go file.line":129},"message":"kubernetes: Node gke-thb-dev-default-pool-b2e32065-fzrr discovered by in cluster pod node query ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.950Z log.logger":"composable.providers.kubernetes origin":{"file.name":"kubernetes/kubernetes.go file.line":113},"message":"Kubernetes provider started for resource node with node scope ecs.version":"1.6.0"}
info 2023-01-17T13:48:00.950Z log.logger":"composable.providers.kubernetes origin":{"file.name":"kubernetes/util.go file.line":122},"message":"kubernetes: Using node gke-thb-dev-default-pool-b2e32065-fzrr provided in the config ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.341Z origin":{"file.name":"coordinator/coordinator.go file.line":729},"message":"Updating running component model ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.342Z origin":{"file.name":"coordinator/coordinator.go file.line":825},"message":"Spawned new component fleet-server-default: Starting: spawned pid '945' component":{"id":"fleet-server-default state":"STARTING"},"ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.343Z origin":{"file.name":"coordinator/coordinator.go file.line":825},"message":"Spawned new unit fleet-server-default-fleet-server: Starting: spawned pid '945' component":{"id":"fleet-server-default state":"STARTING"},"unit":{"id":"fleet-server-default-fleet-server type":"input state":"STARTING"},"ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.343Z origin":{"file.name":"coordinator/coordinator.go file.line":825},"message":"Spawned new unit fleet-server-default: Starting: spawned pid '945' component":{"id":"fleet-server-default state":"STARTING"},"unit":{"id":"fleet-server-default type":"output state":"STARTING"},"ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.543Z origin":{"file.name":"coordinator/coordinator.go file.line":729},"message":"Updating running component model ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.747Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"level":"info time":"2023-01-17T13:48:01Z ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.747Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"level":"info time":"2023-01-17T13:48:01Z ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.747Z message":"Boot fleet-server component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"args":["--agent-mode -E logging.level=debug -E logging.to_stderr=true -E path.data=/usr/share/elastic-agent/state/data/run/fleet-server-default"],"ecs.version":"1.6.0 version":"8.6.0 commit":"05088c13 pid":945,"service.name":"fleet-server ppid":931,"exe":"/usr/share/elastic-agent/data/elastic-agent-b79a5d/components/fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:01.841Z message":"starting communication connection back to Elastic Agent component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
I0117 13:48:01.845453     931 leaderelection.go:248] attempting to acquire leader lease default/elastic-agent-cluster-leader...
I0117 13:48:01.855543     931 leaderelection.go:258] successfully acquired lease default/elastic-agent-cluster-leader
info 2023-01-17T13:48:01.944Z origin":{"file.name":"coordinator/coordinator.go file.line":831},"message":"Component state changed fleet-server-default (STARTING->HEALTHY): Healthy: communicating with pid '945' component":{"id":"fleet-server-default state":"HEALTHY old_state":"STARTING"},"ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.043Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"service.name":"fleet-server ecs.version":"1.6.0 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.043Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.043Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.043Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.043Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.044Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.044Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"service.name":"fleet-server ecs.version":"1.6.0 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.045Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.045Z message":"Setting cache config options component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"service.name":"fleet-server cfg":{"APIKeyJitter":300000000000,"APIKeyTTL":900000000000,"ActionTTL":300000000000,"ArtifactTTL":86400000000000,"EnrollKeyTTL":60000000000,"MaxCost":52428800,"NumCounters":500000},"ecs.version":"1.6.0 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.141Z message":"Starting component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"state":"STARTING ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.141Z message":"No applicable limit for 0 agents, using default. component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"service.name":"fleet-server ecs.version":"1.6.0 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.143Z message":"initial server configuration component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server new":{"Fleet":{"Agent":{"ID":" Logging":{"Level":"info"},"Version":""},"Host":{"ID":" Name":""}},"HTTP":{"Enabled":false,"Host":"localhost Port":5066,"SecurityDescriptor":" User":""},"Inputs":[{"Cache":{"APIKeyJitter":0,"APIKeyTTL":0,"ActionTTL":0,"ArtifactTTL":0,"EnrollKeyTTL":0,"MaxCost":0,"NumCounters":0},"Monitor":{"FetchSize":0,"PollTimeout":0},"Policy":{"ID":""},"Server":{"Bulk":{"FlushInterval":250000000,"FlushMaxPending":8,"FlushThresholdCount":2048,"FlushThresholdSize":1048576},"CompressionLevel":1,"CompressionThresh":1024,"GC":{"CleanupAfterExpiredInterval":"30d ScheduleInterval":3600000000000},"Host":"0.0.0.0 Instrumentation":{"APIKey":" Enabled":false,"Environment":" GlobalLabels":" Hosts":null,"SecretToken":" TLS":{"ServerCA":" ServerCertificate":" SkipVerify":false},"TransactionSampleRate":""},"InternalPort":8221,"Limits":{"AckLimit":{"Burst":100,"Interval":10000000,"Max":50,"MaxBody":2097152},"ArtifactLimit":{"Burst":25,"Interval":5000000,"Max":50,"MaxBody":0},"CheckinLimit":{"Burst":1000,"Interval":1000000,"Max":0,"MaxBody":1048576},"EnrollLimit":{"Burst":100,"Interval":10000000,"Max":50,"MaxBody":524288},"MaxAgents":0,"MaxConnections":52428800,"MaxHeaderByteSize":8192,"PolicyThrottle":500000,"StatusLimit":{"Burst":25,"Interval":5000000,"Max":50,"MaxBody":0}},"Port":8220,"Profiler":{"Bind":"localhost:6060 Enabled":false},"Runtime":{"GCPercent":0},"TLS":{"CASha256":null,"CAs":null,"Certificate":{"Certificate":"/usr/share/fleet-server/config/http-certs/tls.crt Key":"[redacted] Passphrase":""},"CipherSuites":null,"ClientAuth":0,"CurveTypes":null,"Enabled":null,"VerificationMode":"full Versions":null},"Timeouts":{"CheckinJitter":30000000000,"CheckinLongPoll":300000000000,"CheckinTimestamp":30000000000,"Idle":30000000000,"Read":60000000000,"ReadHeader":5000000000,"Write":600000000000}},"Type":""}],"Logging":{"Files":null,"Level":"info Pretty":false,"ToFiles":true,"ToStderr":true},"Output":{"Elasticsearch":{"APIKey":" Headers":null,"Hosts":["elasticsearch-es-http.default.svc:9200"],"MaxConnPerHost":128,"MaxContentLength":104857600,"MaxRetries":3,"Path":" Protocol":"https ProxyDisable":false,"ProxyHeaders":{},"ProxyURL":" ServiceToken":"[redacted] TLS":{"CASha256":null,"CATrustedFingerprint":" CAs":["/mnt/elastic-internal/elasticsearch-association/default/elasticsearch/certs/ca.crt"],"Certificate":{"Certificate":" Key":" Passphrase":""},"CipherSuites":null,"CurveTypes":null,"Enabled":null,"Renegotiation":"never VerificationMode":"full Versions":null},"Timeout":90000000000},"Extra":null}},"ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.143Z message":"starting server on configuration change component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.416Z message":"elasticsearch cluster info component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"cluster.addr":["elasticsearch-es-http.default.svc:9200"],"cluster.maxConnsPersHost":128,"cluster.name":"elasticsearch cluster.uuid":"Pcd7DOp5QxWA7SABzLqvYg ecs.version":"1.6.0 service.name":"fleet-server cluster.version":"8.6.0 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.416Z message":"Run bulker with options component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"service.name":"fleet-server opts":{"apikeyMaxParallel":120,"apikeyMaxReqSize":104857600,"blockQueueSz":32,"flushInterval":250,"flushThresholdCnt":2048,"flushThresholdSz":1048576,"maxPending":8},"ecs.version":"1.6.0 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.419Z message":"Elasticsearch compatibility check successful component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server fleet_version":"8.6.0 elasticsearch_version":"8.6.0 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.480Z message":"elasticsearch cluster info component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"cluster.maxConnsPersHost":128,"cluster.uuid":"Pcd7DOp5QxWA7SABzLqvYg cluster.version":"8.6.0 cluster.addr":["elasticsearch-es-http.default.svc:9200"],"cluster.name":"elasticsearch 2023-01-17T13:48:02.48Z ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.481Z message":"starting index monitor component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server index":".fleet-policies ctx":"index monitor ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.481Z message":"run policy monitor component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"throttle":0.5,"ecs.version":"1.6.0 service.name":"fleet-server ctx":"policy agent monitor ecs.version":"1.6.0"}
warn 2023-01-17T13:48:02.481Z message":"missing config fleet.agent.id; acceptable until Elastic Agent has enrolled component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server ctx":"policy leader manager ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"server listening component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server bind":"0.0.0.0:8220 rdTimeout":60000,"wrTimeout":600000,"ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"fleet-server creating new limiter component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"0.0.0.0:8220 limits":{"AckLimit":{"Burst":100,"Interval":10000000,"Max":50,"MaxBody":2097152},"ArtifactLimit":{"Burst":25,"Interval":5000000,"Max":50,"MaxBody":0},"CheckinLimit":{"Burst":1000,"Interval":1000000,"Max":0,"MaxBody":1048576},"EnrollLimit":{"Burst":100,"Interval":10000000,"Max":50,"MaxBody":524288},"MaxAgents":0,"MaxConnections":52428800,"MaxHeaderByteSize":8192,"PolicyThrottle":500000,"StatusLimit":{"Burst":25,"Interval":5000000,"Max":50,"MaxBody":0}},"ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"starting index monitor component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server index":".fleet-actions ctx":"index monitor ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"service.name":"fleet-server addr":"0.0.0.0:8220 method":"GET path":"/api/status ecs.version":"1.6.0 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"method":"POST path":"/api/fleet/agents/:id ecs.version":"1.6.0 service.name":"fleet-server addr":"0.0.0.0:8220 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"method":"POST path":"/api/fleet/agents/:id/checkin ecs.version":"1.6.0 service.name":"fleet-server addr":"0.0.0.0:8220 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"0.0.0.0:8220 method":"POST path":"/api/fleet/agents/:id/acks ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"0.0.0.0:8220 method":"GET path":"/api/fleet/artifacts/:id/:sha2 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"fleet-server routes set up component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"0.0.0.0:8220 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.541Z message":"server hard connection limiter installed component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server hardConnLimit":52428800,"ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"server listening component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"wrTimeout":600000,"ecs.version":"1.6.0 service.name":"fleet-server bind":"localhost:8221 rdTimeout":60000,"ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"fleet-server creating new limiter component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"localhost:8221 limits":{"AckLimit":{"Burst":100,"Interval":10000000,"Max":50,"MaxBody":2097152},"ArtifactLimit":{"Burst":25,"Interval":5000000,"Max":50,"MaxBody":0},"CheckinLimit":{"Burst":1000,"Interval":1000000,"Max":0,"MaxBody":1048576},"EnrollLimit":{"Burst":100,"Interval":10000000,"Max":50,"MaxBody":524288},"MaxAgents":0,"MaxConnections":52428800,"MaxHeaderByteSize":8192,"PolicyThrottle":500000,"StatusLimit":{"Burst":25,"Interval":5000000,"Max":50,"MaxBody":0}},"ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"localhost:8221 method":"GET path":"/api/status ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"addr":"localhost:8221 method":"POST path":"/api/fleet/agents/:id ecs.version":"1.6.0 service.name":"fleet-server ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"localhost:8221 method":"POST path":"/api/fleet/agents/:id/checkin ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"localhost:8221 method":"POST path":"/api/fleet/agents/:id/acks ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"fleet-server route added component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"localhost:8221 method":"GET path":"/api/fleet/artifacts/:id/:sha2 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"fleet-server routes set up component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server addr":"localhost:8221 ecs.version":"1.6.0"}
info 2023-01-17T13:48:02.542Z message":"server hard connection limiter installed component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server hardConnLimit":52428800,"ecs.version":"1.6.0"}
info 2023-01-17T13:48:03.125Z message":"Running on policy with Fleet Server integration: eck-fleet-server; missing config fleet.agent.id (expected during bootstrap process) component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server state":"DEGRADED ecs.version":"1.6.0"}
warn 2023-01-17T13:48:03.125Z origin":{"file.name":"coordinator/coordinator.go file.line":829},"message":"Unit state changed fleet-server-default-fleet-server (STARTING->DEGRADED): Running on policy with Fleet Server integration: eck-fleet-server; missing config fleet.agent.id (expected during bootstrap process) component":{"id":"fleet-server-default state":"HEALTHY"},"unit":{"id":"fleet-server-default-fleet-server type":"input state":"DEGRADED old_state":"STARTING"},"ecs.version":"1.6.0"}
warn 2023-01-17T13:48:03.125Z origin":{"file.name":"coordinator/coordinator.go file.line":829},"message":"Unit state changed fleet-server-default (STARTING->DEGRADED): Running on policy with Fleet Server integration: eck-fleet-server; missing config fleet.agent.id (expected during bootstrap process) component":{"id":"fleet-server-default state":"HEALTHY"},"unit":{"id":"fleet-server-default type":"output state":"DEGRADED old_state":"STARTING"},"ecs.version":"1.6.0"}
info 2023-01-17T13:48:04.248Z origin":{"file.name":"cmd/enroll_cmd.go file.line":784},"message":"Fleet Server - Running on policy with Fleet Server integration: eck-fleet-server; missing config fleet.agent.id (expected during bootstrap process) ecs.version":"1.6.0"}
info 2023-01-17T13:48:05.176Z origin":{"file.name":"cmd/enroll_cmd.go file.line":475},"message":"Starting enrollment to URL: https://fleet-server-agent-http.default.svc:8220/ ecs.version":"1.6.0"}
info 2023-01-17T13:48:05.248Z message":"Elastic Agent successfully enrolled component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"http.request.id":"01GPZZS9S11BCSZ5X8HXMFWR7V http.response.body.bytes":1495,"event.duration":980491236,"ecs.version":"1.6.0 fleet.enroll.apikey.id":"C3_7v4UB1L2vmVtd_qDp fleet.agent.id":"996ab694-ef86-4f09-9ad3-47b7cf2fa1ed fleet.policy.id":"eck-agent fleet.access.apikey.id":"O-X8v4UBc8M4w3BNqMGN service.name":"fleet-server mod":"enroll ecs.version":"1.6.0"}
info 2023-01-17T13:48:06.039Z message":"Elastic Agent successfully enrolled component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 http.request.id":"01GPZZSAY3PXZ55A421HCJ92T1 fleet.agent.id":"d3bd6dd0-3715-41df-ae8f-e05869c82422 fleet.policy.id":"eck-fleet-server http.response.body.bytes":1012,"event.duration":586055123,"service.name":"fleet-server mod":"enroll fleet.enroll.apikey.id":"N-X7v4UBc8M4w3BN_sHx fleet.access.apikey.id":"PeX8v4UBc8M4w3BNrMHa ecs.version":"1.6.0"}
Successfully enrolled the Elastic Agent.
info 2023-01-17T13:48:06.259Z origin":{"file.name":"cmd/enroll_cmd.go file.line":277},"message":"Elastic Agent has been enrolled; start Elastic Agent ecs.version":"1.6.0"}
info 2023-01-17T13:48:06.259Z origin":{"file.name":"cmd/run.go file.line":242},"message":"signal \"terminated\" received ecs.version":"1.6.0"}
info 2023-01-17T13:48:06.259Z origin":{"file.name":"cmd/run.go file.line":254},"message":"Shutting down Elastic Agent and sending last events... ecs.version":"1.6.0"}
info 2023-01-17T13:48:06.259Z message":"On signal component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server sig":"terminated ecs.version":"1.6.0"}
info 2023-01-17T13:48:06.341Z message":"Stopping component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"ecs.version":"1.6.0 service.name":"fleet-server state":"STOPPING 2023-01-17T13:48:06.26Z ecs.version":"1.6.0"}
error 2023-01-17T13:48:06.341Z message":"error while closing server component":{"binary":"fleet-server dataset":"elastic_agent.fleet_server id":"fleet-server-default type":"fleet-server"},"@timestamp":"2023-01-17T13:48:06.26Z ecs.version":"1.6.0 service.name":"fleet-server error.message":"close tcp 127.0.0.1:8221: use of closed network connection ecs.version":"1.6.0"}
```

</details>

--- a/test/e2e/agent/config_test.go
+++ b/test/e2e/agent/config_test.go
@@ -124,10 +124,13 @@ func TestMultipleOutputConfig(t *testing.T) {
 }
 
 func TestFleetMode(t *testing.T) {
-
-	test.SkipUntilResolution(t, 6331)
-
 	v := version.MustParse(test.Ctx().ElasticStackVersion)
+
+	// https://github.com/elastic/cloud-on-k8s/issues/6331
+	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+		t.SkipNow()
+	}
+
 	// installation of policies and integrations through Kibana file based configuration was broken between those versions:
 	if v.LT(version.MinFor(8, 1, 0)) && v.GTE(version.MinFor(8, 0, 0)) {
 		t.SkipNow()

--- a/test/e2e/agent/config_test.go
+++ b/test/e2e/agent/config_test.go
@@ -127,7 +127,7 @@ func TestFleetMode(t *testing.T) {
 	v := version.MustParse(test.Ctx().ElasticStackVersion)
 
 	// https://github.com/elastic/cloud-on-k8s/issues/6331
-	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+	if v.LT(version.MinFor(8, 7, 0)) && v.GE(version.MinFor(8, 6, 0)) {
 		t.SkipNow()
 	}
 

--- a/test/e2e/agent/recipes_test.go
+++ b/test/e2e/agent/recipes_test.go
@@ -88,8 +88,12 @@ func TestMultiOutputRecipe(t *testing.T) {
 }
 
 func TestFleetKubernetesIntegrationRecipe(t *testing.T) {
+	v := version.MustParse(test.Ctx().ElasticStackVersion)
 
-	test.SkipUntilResolution(t, 6331)
+	// https://github.com/elastic/cloud-on-k8s/issues/6331
+	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+		t.SkipNow()
+	}
 
 	customize := func(builder agent.Builder) agent.Builder {
 		if !builder.Agent.Spec.FleetServerEnabled {
@@ -133,8 +137,12 @@ func TestFleetKubernetesIntegrationRecipe(t *testing.T) {
 }
 
 func TestFleetCustomLogsIntegrationRecipe(t *testing.T) {
+	v := version.MustParse(test.Ctx().ElasticStackVersion)
 
-	test.SkipUntilResolution(t, 6331)
+	// https://github.com/elastic/cloud-on-k8s/issues/6331
+	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+		t.SkipNow()
+	}
 
 	notLoggingPod := beat.NewPodBuilder("test")
 	loggingPod := beat.NewPodBuilder("test")
@@ -162,8 +170,12 @@ func TestFleetCustomLogsIntegrationRecipe(t *testing.T) {
 }
 
 func TestFleetAPMIntegrationRecipe(t *testing.T) {
+	v := version.MustParse(test.Ctx().ElasticStackVersion)
 
-	test.SkipUntilResolution(t, 6331)
+	// https://github.com/elastic/cloud-on-k8s/issues/6331
+	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+		t.SkipNow()
+	}
 
 	customize := func(builder agent.Builder) agent.Builder {
 		if !builder.Agent.Spec.FleetServerEnabled {

--- a/test/e2e/agent/recipes_test.go
+++ b/test/e2e/agent/recipes_test.go
@@ -91,7 +91,7 @@ func TestFleetKubernetesIntegrationRecipe(t *testing.T) {
 	v := version.MustParse(test.Ctx().ElasticStackVersion)
 
 	// https://github.com/elastic/cloud-on-k8s/issues/6331
-	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+	if v.LT(version.MinFor(8, 7, 0)) && v.GE(version.MinFor(8, 6, 0)) {
 		t.SkipNow()
 	}
 
@@ -140,7 +140,7 @@ func TestFleetCustomLogsIntegrationRecipe(t *testing.T) {
 	v := version.MustParse(test.Ctx().ElasticStackVersion)
 
 	// https://github.com/elastic/cloud-on-k8s/issues/6331
-	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+	if v.LT(version.MinFor(8, 7, 0)) && v.GE(version.MinFor(8, 6, 0)) {
 		t.SkipNow()
 	}
 
@@ -173,7 +173,7 @@ func TestFleetAPMIntegrationRecipe(t *testing.T) {
 	v := version.MustParse(test.Ctx().ElasticStackVersion)
 
 	// https://github.com/elastic/cloud-on-k8s/issues/6331
-	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+	if v.LT(version.MinFor(8, 7, 0)) && v.GE(version.MinFor(8, 6, 0)) {
 		t.SkipNow()
 	}
 

--- a/test/e2e/agent/tls_test.go
+++ b/test/e2e/agent/tls_test.go
@@ -18,10 +18,12 @@ import (
 
 // TestFleetAgentWithoutTLS tests a Fleet Server, and Elastic Agent with TLS disabled for the HTTP layer.
 func TestFleetAgentWithoutTLS(t *testing.T) {
-
-	test.SkipUntilResolution(t, 6331)
-
 	v := version.MustParse(test.Ctx().ElasticStackVersion)
+
+	// https://github.com/elastic/cloud-on-k8s/issues/6331
+	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+		t.SkipNow()
+	}
 
 	// Disabling TLS for Fleet isn't supported before 7.16, as Elasticsearch doesn't allow
 	// api keys to be enabled when TLS is disabled.

--- a/test/e2e/agent/tls_test.go
+++ b/test/e2e/agent/tls_test.go
@@ -21,7 +21,7 @@ func TestFleetAgentWithoutTLS(t *testing.T) {
 	v := version.MustParse(test.Ctx().ElasticStackVersion)
 
 	// https://github.com/elastic/cloud-on-k8s/issues/6331
-	if v.LT(version.MinFor(8, 8, 0)) && v.GE(version.MinFor(8, 6, 0)) {
+	if v.LT(version.MinFor(8, 7, 0)) && v.GE(version.MinFor(8, 6, 0)) {
 		t.SkipNow()
 	}
 

--- a/test/e2e/agent/upgrade_test.go
+++ b/test/e2e/agent/upgrade_test.go
@@ -9,6 +9,7 @@ package agent
 import (
 	"testing"
 
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/version"
 	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test"
 	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test/agent"
 	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test/elasticsearch"
@@ -17,11 +18,20 @@ import (
 
 func TestAgentVersionUpgradeToLatest8x(t *testing.T) {
 
-	test.SkipUntilResolution(t, 6331)
-
 	srcVersion, dstVersion := test.GetUpgradePathTo8x(test.Ctx().ElasticStackVersion)
 
 	test.SkipInvalidUpgrade(t, srcVersion, dstVersion)
+
+	sv := version.MustParse(srcVersion)
+	dv := version.MustParse(dstVersion)
+
+	// https://github.com/elastic/cloud-on-k8s/issues/6331
+	if sv.LT(version.MinFor(8, 8, 0)) && sv.GE(version.MinFor(8, 6, 0)) {
+		t.SkipNow()
+	}
+	if dv.LT(version.MinFor(8, 8, 0)) && dv.GE(version.MinFor(8, 6, 0)) {
+		t.SkipNow()
+	}
 
 	name := "test-agent-upgrade"
 	esBuilder := elasticsearch.NewBuilder(name).

--- a/test/e2e/agent/upgrade_test.go
+++ b/test/e2e/agent/upgrade_test.go
@@ -26,10 +26,10 @@ func TestAgentVersionUpgradeToLatest8x(t *testing.T) {
 	dv := version.MustParse(dstVersion)
 
 	// https://github.com/elastic/cloud-on-k8s/issues/6331
-	if sv.LT(version.MinFor(8, 8, 0)) && sv.GE(version.MinFor(8, 6, 0)) {
+	if sv.LT(version.MinFor(8, 7, 0)) && sv.GE(version.MinFor(8, 6, 0)) {
 		t.SkipNow()
 	}
-	if dv.LT(version.MinFor(8, 8, 0)) && dv.GE(version.MinFor(8, 6, 0)) {
+	if dv.LT(version.MinFor(8, 7, 0)) && dv.GE(version.MinFor(8, 6, 0)) {
 		t.SkipNow()
 	}
 

